lib/rouge/lexers/solidity.rb

```diff
@@ -0,0 +1,179 @@
+      mimetypes 'text/x-solidity'
+
+      # optional comment or whitespace
+      ws = %r((?:\s|//.*?\n|/[*].*?[*]/)+)
+      id = /[a-zA-Z$_][\w$_]*/
+
+      def self.detect?(text)
+        return true if text.start_with? 'pragma solidity'
+      end
+
+      # TODO: seperate by "type"
+      def self.keywords
+        @keywords ||= Set.new %w(
+          abstract anonymous as assembly break catch calldata constant
+          constructor continue contract do delete else emit enum event
+          external fallback for function hex if indexed interface
+          internal import is library mapping memory modifier new
+          override payable public pure pragma private receive return
+          returns storage struct throw try type using var view virtual
+          while
+        )
+      end
+
+      def self.builtins
+        return @builtins if @builtins
+
+        @builtins = Set.new %w(
+          now
+          false true
+          balance now selector super this
+          blockhash gasleft
+          assert require revert
+          selfdestruct suicide
+          call callcode delegatecall
+          send transfer
+          addmod ecrecover keccak256 mulmod sha256 sha3 ripemd160
+        )
+
+        # TODO: use (currently shadowed by catch-all in :statements)
+        abi = %w(encode encodePacked encodeWithSelector encodeWithSignature)
+        @builtins.merge( abi.map { |i| "abi.#{i}" } )
+        block = %w(coinbase difficulty gaslimit hash number timestamp)
+        @builtins.merge( block.map { |i| "block.#{i}" } )
+        msg = %w(data gas sender sig value)
+        @builtins.merge( msg.map { |i| "msg.#{i}" } )
+        tx = %w(gasprice origin)
+        @builtins.merge( tx.map { |i| "tx.#{i}" } )
+      end
+
+      def self.constants
+        @constants ||= Set.new %w(
+          wei finney szabo ether
+          seconds minutes hours days weeks years
+        )
+      end
+
+      def self.keywords_type
+        @keywords_type ||= Set.new %w(
+          address bool byte bytes int string uint
+        )
+      end
+
+      def self.reserved
+        @reserved ||= Set.new %w(
+          alias after apply auto case copyof default define final fixed
+          immutable implements in inline let macro match mutable null of
+          partial promise reference relocatable sealed sizeof static
+          supports switch typedef typeof ufixed unchecked
+        )
+      end
+
+      start { push :bol }
+
+      state :expr_bol do
+        mixin :inline_whitespace
+
+        rule(//) { pop! }
+      end
+
+      # :expr_bol is the same as :bol but without labels, since
+      # labels can only appear at the beginning of a statement.
+      state :bol do
+        mixin :expr_bol
+      end
+
+      # TODO: natspec in comments
+      state :inline_whitespace do
+        rule %r/[ \t\r]+/, Text
+        rule %r/\\\n/, Text # line continuation
+        rule %r(/(\\\n)?[*].*?[*](\\\n)?/)m, Comment::Multiline
```

Comment:
When I look at the visual sample, this code:

```
 // These would close the comment if the space was removed:
    * /
    \* /
```

results in the `\` being marked with an error token. Is that what's supposed to happen?

Reply:
Nope! I think it's happening because the single-line comments above close out the multi-line comment. Trying to fix now...

Reply:
Without thinking too much, I added the case of unclosed multi-line comment to be considered one, too: https://github.com/rouge-ruby/rouge/pull/760/commits/f19341e7ff80612886e4febfdea211f7c0b36656